ASM/Makefile

```diff
@@ -28,10 +33,22 @@ endif
 $(OBJDIR): 
 	mkdir -p $@
 
+$(OBJDIR)/%_bin.o: $(RESOURCEDIR)/%.bin
+	$(OBJCOPY) -I binary -O elf32-bigmips --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
+	$(OBJCOPY) --redefine-sym _binary_resources_$*_bin_start=$(call UC,$*)_RESOURCE $@
+	$(OBJCOPY) --redefine-sym _binary_resources_$*_bin_end=$(call UC,$*)_RESOURCE_END $@
+	$(OBJCOPY) --redefine-sym _binary_resources_$*_bin_size=$(call UC,$*)_RESOURCE_SIZE $@
+
+
+$(RESOURCEDIR):
+	mkdir -p $@
+
 $(OBJECTS): | $(OBJDIR)
 
-bundle: $(OBJECTS)
-	$(LD) -o $(OUTDIR)/bundle.o -i -L. $(patsubst %.o,-l:%.o,$(OBJECTS))
+$(RESOURCES): | $(RESOURCEDIR)
+
+bundle: $(RESOURCES) $(OBJECTS)
+	$(LD) -o $(OUTDIR)/bundle.o -i -L. $(patsubst %.o,-l:%.o,$(OBJECTS) $(RESOURCES))
```

Comment:
Does any object depend on any resource already being built before building said object?

Reply:
I definitely don't fully understand the question. The symbols for the resource just have to be created and linked so that the objects can see them. I'm not sure if that's the "resource already being built before building said object"

To be clear, I'm also not great at Makefiles. All I know is that my idea seemed to work here. It's very possible I either did something wrong, or in a hard to read/understand way.

Reply:
I mean if any of the built objects depend on any of the resources being built already.
For example, say `example.o` requires `texture_bin.o` to be already existing otherwise it can't be built. If that happened then you end up introducing a race condition to the build system.

By re-reading the changes of the PR it doesn't seem like either objects or resources depend on each other tho